### PR TITLE
add manage user page

### PR DESCRIPTION
**Pull request by 98ZhaoJeffrey**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/7e51fde5f22d448b8226f2cfa4dd93d5?v=b2dc0dba06574651a8b6063bdd0d74e0&p=6bddda91d13842c3951f993ffbe5eea9&pm=s)

## Implementation description
* Added a page /manage-users where you can see all users based on a certain role
* Made it such that it paginates to overwhelm people when there are lots of users (prolly won't happen)

## Steps to test
1. In an admin account, go to /manage-users
2. By default it should display the admin users in the table
3. Changing the selected role should also change the users displayed
4. There is a footer for the table to adjust number of users per page as well as changing the page. Since I nuked the database's users, theres only so many, so you can try manually changing the number of rows per page and see if it works correctly that way

## What should reviewers focus on?
* Check how it looks when we have more user/data filling out the table
* Check if the aria-labels and design complies with A11Y

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by ji-mmyliu — APPROVED** [review]
LGTM
* Tested for unauthorized access to `/manage-users`
* Tested with adding more users

File: frontend/src/components/pages/ManageUserPage.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                <TableCell style={{ width: 160 }} align="right">
+                  {user.firstName}
+                </TableCell>
+                <TableCell style={{ width: 160 }} align="right">
+                  {user.lastName}
+                </TableCell>
+              </TableRow>
+            ))}
+            {emptyRows > 0 && (
+              <TableRow style={{ height: 25 * emptyRows }}>
```
Comment (carolynzhang18) [inline frontend/src/components/pages/ManageUserPage.tsx:102]: this'll make one tall empty row - did you mean to make several empty rows, like a `TableRow` for each empty row?

**Review by carolynzhang18 — APPROVED** [review]
image

Love the pagination right off the bat, works super nicely! 🕺

File: frontend/src/components/pages/ManageUserPage.tsx (hunk 1 above)
Comment (98ZhaoJeffrey) [reply frontend/src/components/pages/ManageUserPage.tsx:102]: The I just copied from this https://mui.com/material-ui/react-table/

Does it make more sense to create 1 big row or multiple rows to fill out the empty space?

Comment (carolynzhang18) [reply frontend/src/components/pages/ManageUserPage.tsx:102]: Ohhh ok I get what you mean... you're right, 1 big empty space would be better.

Was the whole `emptyRows` thing also something you got from MUI docs? I was looking at https://mui.com/material-ui/react-table/#data-table and it seems like the empty space comes with the component already

Comment (98ZhaoJeffrey) [reply frontend/src/components/pages/ManageUserPage.tsx:102]: I think I copied the one under custom pagination options. Should I use that one or the one you linked?

Comment (carolynzhang18) [reply frontend/src/components/pages/ManageUserPage.tsx:102]: Hmm we can leave it like this for now, and then once designers make designs for this page we can revisit if needed. What do you think?

I remember noticing a change in height between page 1 and page 2 of Facilitators @ 5 per page when I was testing last night, so I wasn't sure if it's a bug... but if that's how MUI had it we can leave it at that and polish when we implement designs. All good as far as this ticket goes either way! Thanks 😎

Comment (98ZhaoJeffrey) [reply frontend/src/components/pages/ManageUserPage.tsx]: ok so basically the data table looked too scary and I didn't want to do it the way they did it so I just copied how it was done in the customize seciton. I changed it so every row is equal height and stuff

**Pull request merged**